### ISSUE TITLE
`addAutoImport` and `addAutoImportDir` are not provided by `@nuxt/kit`

### ISSUE THREAD

**Issue by agracia-foticos** [nuxt/framework#9063]
### Environment

- Operating System: `Linux`
- Node Version:     `v18.11.0`
- Nuxt Version:     `3.0.0-rc.14`
- Nitro Version:    `1.0.0`
- Package Manager:  `npm@8.19.2`
- Builder:          `vite`
- User Config:      `srcDir`, `modules`, `compilerOptions`, `vue`, `vite`, `schemaOrg`, `telemetry`, `algolia`, `head`, `auth`, `css`, `build`, `runtimeConfig`
- Runtime Modules:  `genesis-design-system/nuxt`, `@nuxtjs/tailwindcss@6.1.3`, `@nuxtjs/algolia@1.1.5`, `@nuxtjs-alt/auth@2.1.3`, `@nuxtjs-alt/http@1.4.6`, `@nuxtjs/device@3.0.0`, `@nuxtjs/partytown@1.2.2`, `@vueuse/nuxt@9.5.0`, `nuxt-schema-org@1.1.0-beta.6`, `unplugin-icons/nuxt`, `@pinia/nuxt@0.4.3`, `@nuxtjs/i18n@8.0.0-beta.4`
- Build Modules:    `-`

### Reproduction

I have upgraded from RC 13 to RC 14 and the server doesn't even start.

![image](https://user-images.githubusercontent.com/77386602/202210879-4e86caf2-7cab-444b-8b35-9348d286f00c.png)

### Describe the bug

I have upgraded from RC 13 to RC 14 and the server doesn't even start.

### Logs

```shell
ERROR  Error while requiring module @nuxtjs/algolia: SyntaxError: The requested module '@nuxt/kit' does not provide an export named 'addAutoImportDir'


 ERROR  Cannot restart nuxt:  The requested module '@nuxt/kit' does not provide an export named 'addAutoImportDir'

  import { defineNuxtModule, addComponentsDir, addPlugin, addAutoImportDir, addServerHandler } from '@nuxt/kit';
  ^^^^^^^^^^^^^^^^
  SyntaxError: The requested module '@nuxt/kit' does not provide an export named 'addAutoImportDir'
  at ModuleJob._instantiate (node:internal/modules/esm/module_job:123:21)
  at async ModuleJob.run (node:internal/modules/esm/module_job:189:5)
  at async Promise.all (index 0)
  at async ESMLoader.import (node:internal/modules/esm/loader:526:24)
  at async normalizeModule (node_modules/@nuxt/kit/dist/index.mjs:432:28)
  at async installModule (node_modules/@nuxt/kit/dist/index.mjs:415:41)
  at async initNuxt (node_modules/nuxt/dist/index.mjs:1827:7)
  at async load (node_modules/nuxi/dist/chunks/dev.mjs:6779:9)
  at async Object.invoke (node_modules/nuxi/dist/chunks/dev.mjs:6840:5)
  at async _main (node_modules/nuxi/dist/cli.mjs:50:20)
```

**Comment by existe-deja** [comment]
I confirm the bug for other modules like https://github.com/nuxt-modules/tailwindcss with an error like
```bash
FATAL  Cannot destructure property 'nuxt' of 'this' as it is undefined.
```
For yarn users, adding 
```
  "resolutions": {
    "@nuxt/kit": "3.0.0-rc.13"
  }
  ```
  fix the bug.

**Comment by toniengelhardt** [comment]
Same with WindiCSS

![Screenshot 2022-11-16 at 14 56 58](https://user-images.githubusercontent.com/2703233/202215054-a43a14e4-4b37-49ff-a2e5-3769434c1361.png)

**Comment by gravit8** [comment]
Also had issues with RC14 and a fresh RC13, added "@nuxt/kit": "3.0.0-rc.13" to resolve RC13 for now.

**Comment by kosmeln** [comment]
Same here - different modules though (font awesome). But it probably doesn't matter... Any help there?

**Comment by toniengelhardt** [comment]
> Same here - different modules though (font awesome). But it probably doesn't matter... Any help there?

For WindiCSS I solved it by adding 

```ts
{
  buildModules: []
}
```

to `nuxt.config.ts`. Maybe that also works for fa.

**Comment by kosmeln** [comment]
thanks a ton @toniengelhardt !

**Comment by Corentin7301** [comment]
Same error when I've uploaded to rc14 bug the same `The requested module '@nuxt/kit' does not provide an export named 'addAutoImport'` error is always here when I downgrade to rc13...
I use pnpm, I've deleted node-modules & pnpm-lock before execute `pnpm install --shamefully-hoist`.
Have you a bypass solution?

**Comment by danielroe** [comment]
@Corentin7301 It seems you have a module using a deprecated (and now removed) name. It might be caused by any number of modules. If you can find out which it is, feel free to raise an issue in that repo and tag me; I'll happily resolve.

**Comment by Corentin7301** [comment]
@danielroe Okay, thanks for your answer.
I'll ckecked towmorrow. If I found the module, I'll shared here.
By chance, it would not be the image module (for Nuxt 3)? (Because I use image module in my Nuxt 3 app and it is an official module).
Thanks!

**Comment by Sandlens** [comment]
> @danielroe Okay, thanks for your answer.
I'll ckecked towmorrow. If I found the module, I'll shared here.
By chance, it would not be the image module (for Nuxt 3)? (Because I use image module in my Nuxt 3 app and it is an official module).
Thanks!

Uninstalling @nuxt/image-edge fixed it for me (although I had locked my version and wasn't up to date with the latest release).

**Comment by cinob** [comment]
Same with vercel
```
[error] Cannot destructure property 'nuxt' of 'this' as it is undefined.
--
10:22:33.336 | at webVitalsModule (node_modules/.pnpm/@nuxtjs+web-vitals@0.1.10/node_modules/@nuxtjs/web-vitals/dist/index.js:47:11)
10:22:33.337 | at installModule (node_modules/.pnpm/@nuxt+kit@3.0.0/node_modules/@nuxt/kit/dist/index.mjs:416:9)
10:22:33.337 | at async initNuxt (node_modules/.pnpm/nuxt@3.0.0_ykiqiucjbh4zyhanmulizxncqi/node_modules/nuxt/dist/index.mjs:1825:7)
10:22:33.337 | at async loadNuxt (node_modules/.pnpm/nuxt@3.0.0_ykiqiucjbh4zyhanmulizxncqi/node_modules/nuxt/dist/index.mjs:1857:5)
10:22:33.337 | at async loadNuxt (node_modules/.pnpm/@nuxt+kit@3.0.0/node_modules/@nuxt/kit/dist/index.mjs:493:19)
10:22:33.337 | at async Object.invoke (node_modules/.pnpm/nuxi@3.0.0/node_modules/nuxi/dist/chunks/build.mjs:34:18)
10:22:33.337 | at async _main (node_modules/.pnpm/nuxi@3.0.0/node_modules/nuxi/dist/cli.mjs:50:20)

**Comment by Corentin7301** [comment]
@Sandlens Do you think there is a solution to still use nuxt-image?

**Comment by agracia-foticos** [comment]
> Also had issues with RC14 and a fresh RC13, added "@nuxt/kit": "3.0.0-rc.13" to dependencies to resolve RC13 for now.


I've set @nuxt/kit to RC13 and it still can't get the server up


`business@nuc-it-java:~/var/www/project$ npm list
nuxt@ /home/business/var/www/nuxt
├── @adyen/adyen-web@5.29.0
├── @analytics/segment@1.1.4
├── @headlessui/vue@1.7.4
├── @iconify-json/fa6-brands@1.1.7
├── @iconify-json/fa6-solid@1.1.8
├── @intlify/nuxt3@0.2.4
├── @mapbox/mapbox-gl-geocoder@5.0.1
├── @nuxt/kit@3.0.0-rc.13
├── @nuxt/postcss8@1.1.3
├── @nuxtjs-alt/auth@2.1.5
├── @nuxtjs-alt/http@1.5.0
├── @nuxtjs/algolia@1.2.0
├── @nuxtjs/auth-next@5.0.0-1667386184.dfbbb54
├── @nuxtjs/device@3.0.0 (git+ssh://git@github.com/Redemption198/device-module.git#2136ea04d9b1281a1d14d1c91ed40641106bbfaf)
├── @nuxtjs/i18n@8.0.0-beta.4
├── @nuxtjs/partytown@1.2.2
├── @nuxtjs/tailwindcss@6.1.3
├── @pinia/nuxt@0.4.4
├── @splidejs/vue-splide@0.6.12
├── @studiometa/vue-mapbox-gl@2.0.1
├── @tailwindcss/typography@0.5.8
├── @vueuse/nuxt@9.5.0
├── analytics@0.8.1
├── autoprefixer@10.4.13
├── chart.js@3.9.1
├─┬ genesis-design-system@0.1.0 -> ./../genesis-design-system
│ ├── @babel/core@7.20.2
│ ├── @babel/eslint-parser@7.19.1
│ ├── @headlessui/vue@1.7.4
│ ├── @heroicons/vue@1.0.6
│ ├── @mdx-js/react@2.1.5
│ ├── @nuxt/kit@3.0.0-rc.13
│ ├── @storybook/addon-actions@6.5.13
│ ├── @storybook/addon-docs@6.5.13
│ ├── @storybook/addon-essentials@6.5.13
│ ├── @storybook/addon-interactions@6.5.13
│ ├── @storybook/addon-links@6.5.13
│ ├── @storybook/addon-postcss@2.0.0
│ ├── @storybook/builder-webpack5@6.5.13
│ ├── @storybook/manager-webpack5@6.5.13
│ ├── @storybook/testing-library@0.0.11
│ ├── @storybook/vue3@6.5.13
│ ├── @tailwindcss/forms@0.5.3
│ ├── @tailwindcss/typography@0.5.8
│ ├── @vue/cli-plugin-babel@5.0.8
│ ├── @vue/cli-plugin-eslint@5.0.8
│ ├── @vue/cli-service@5.0.8
│ ├── autoprefixer@10.4.13
│ ├── babel-loader@8.3.0
│ ├── core-js@3.26.0
│ ├── eslint-plugin-vue@8.7.1
│ ├── eslint@7.32.0
│ ├── postcss-cli@9.1.0
│ ├── postcss-copy@7.1.0
│ ├── postcss-import@14.1.0
│ ├── postcss-important@0.3.0
│ ├── postcss-nested@5.0.6
│ ├── postcss@8.4.18
│ ├── tailwindcss@3.2.2
│ ├── vue-loader@16.8.3
│ └── vue@3.2.41
├── mapbox-gl@2.11.0
├── npx@10.2.2
├── nuxt-schema-org@1.1.0-beta.6
├── nuxt@3.0.0-rc.14
├── pinia@2.0.24
├── postcss@8.4.19
├── redis@4.5.0
├── unplugin-icons@0.14.13
├── vue-chart-3@3.1.8
├── vue-tel-input@6.0.5
├── vue@3.2.45
└── vue3-carousel@0.1.48`

**Comment by Corentin7301** [comment]
I confirm that the error actually comes from the @nuxt/image-edge module.
My actual solution is to remove the module.

**Comment by lavellotron** [comment]
I have the same issue but with `@storyblok/nuxt` and the Nuxt Kit method `addAutoImport` being undefined.

Seems that the Nuxt Kit interface changed and no alias methods were provided for the old interface. I imagine nearly all 3rd-party Nuxt 3 modules are broken right now. And will be until each vendor releases an update. 

I rolled back to `"nuxt": "3.0.0-rc.12"` and `"@nuxt/kit": "3.0.0-rc.12"` so I can continue building my client's website.

**Comment by danielroe** [comment]
We deprecated those helpers quite in RC9 over two months ago and provided alias methods. But of course we've removed them prior to the release and API stability.

**Comment by lavellotron** [comment]
My issue is fixed after updating `@storyblok/nuxt` to `v4.5.0`.
